# **Phase 4: Exploratory Data Analysis (EDA)**

In [4]:
import pandas as pd
df = pd.read_csv('Pakistan_Ecommerce_Dataset_for_EDA.csv')
df

C:\Users\Abdul Basit\AppData\Local\Temp\ipykernel_7760\4110039818.py:2: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('Pakistan_Ecommerce_Dataset_for_EDA.csv')


,item_id,status,created_at,sku,price,qty_ordered,grand_total,increment_id,category_name_1,sales_commission_code,discount_amount,payment_method,Working Date,BI Status,MV,Year,Month,Customer Since,M-Y,FY,Customer ID,day_name,month_name
0,211131,complete,2016-07-01,kreations_yi 06-l,1950.0,1,1950.0,100147443,women's fashion,\N,0.0,cod,2016-07-01,#REF!,"1,950",2016,7,2016-7,7-2016,FY17,1,Friday,July
1,211133,canceled,2016-07-01,kcc_buy 2 frey air freshener & get 1 kasual bo...,240.0,1,240.0,100147444,beauty & grooming,\N,0.0,cod,2016-07-01,Gross,240,2016,7,2016-7,7-2016,FY17,2,Friday,July
2,211134,canceled,2016-07-01,ego_up0017-999-mr0,2450.0,1,2450.0,100147445,women's fashion,\N,0.0,cod,2016-07-01,Gross,"2,450",2016,7,2016-7,7-2016,FY17,3,Friday,July
3,211135,complete,2016-07-01,kcc_krone deal,360.0,1,60.0,100147446,beauty & grooming,R-FSD-52352,300.0,cod,2016-07-01,Net,360,2016,7,2016-7,7-2016,FY17,4,Friday,July
4,211136,order_refunded,2016-07-01,bk7010400ag,555.0,2,1110.0,100147447,soghaat,\N,0.0,cod,2016-07-01,Valid,"1,110",2016,7,2016-7,7-2016,FY17,5,Friday,July
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
584440,905204,cod,2018-08-28,wofsce5ae00357aecde,699.0,1,849.0,100562385,women's fashion,NaN,0.0,cod,2018-08-28,Valid,699,2018,8,2018-8,8-2018,FY19,115320,Tuesday,August
584441,905205,processing,2018-08-28,mathua5af70a7d1e50a,35599.0,1,35899.0,100562386,mobiles & tablets,NaN,0.0,bankalfalah,2018-08-28,Gross,"35,599",2018,8,2018-8,8-2018,FY19,115326,Tuesday,August
584442,905206,processing,2018-08-28,matsam5b6d7208c6d30,129999.0,2,652178.0,100562387,mobiles & tablets,NaN,0.0,bankalfalah,2018-08-28,Gross,"259,998",2018,8,2018-7,8-2018,FY19,113474,Tuesday,August
584443,905207,processing,2018-08-28,matsam5b1509b4696ea,87300.0,2,652178.0,100562387,mobiles & tablets,NaN,0.0,bankalfalah,2018-08-28,Gross,"174,600",2018,8,2018-7,8-2018,FY19,113474,Tuesday,August


In [ ]:
# Checking top 5 Categories by Orders.
print("Top 5 Categories by number of Orders:\n", df['category_name_1'].value_counts().head(5))


Top 5 Categories by number of Orders:
 category_name_1
mobiles & tablets    115701
men's fashion         92217
women's fashion       59717
appliances            52404
superstore            43598
Name: count, dtype: int64


In [6]:
# Payment Method Frequency.
print("Payment Method Distribution:\n", df['payment_method'].value_counts())

# Order Status Frequency.
print('\nOrder Status Distribution:\n', df['status'].value_counts())

Payment Method Distribution:
 payment_method
cod                  271960
payaxis               97617
easypay               82861
jazzwallet            35145
easypay_voucher       31176
bankalfalah           23049
jazzvoucher           15633
easypay_ma            14028
customercredit         7555
apg                    1758
ublcreditcard           882
cashatdoorstep          732
mcblite                 723
mygateway               669
internetbanking         472
productcredit           125
marketingexpense         45
financesettlement        15
Name: count, dtype: int64

Order Status Distribution:
 status
complete          233658
canceled          201229
received           77280
order_refunded     59525
refund              8032
cod                 2859
paid                1159
closed               494
payment_review        57
pending               48
processing            33
holded                31
unknown               19
fraud                 10
pending_paypal         7
exchange      

In [7]:
# Checking correlation of 'Payment Method' and 'Order Status'.
payment_method_corr = pd.crosstab(df['payment_method'], df['status'])
main_status = ['complete', 'canceled', 'order_refunded']
# Filtering Only needed 'Order Statuses'.
print('Correlation: Payment Method VS Order Status')
print(payment_method_corr[main_status])

Correlation: Payment Method VS Order Status
status             complete  canceled  order_refunded
payment_method                                       
apg                      27      1361              42
bankalfalah             570     16184             746
cashatdoorstep          674         5              30
cod                  148039     21753           48771
customercredit         4151        47            1269
easypay               19206     52040            2619
easypay_ma             3116      9210             315
easypay_voucher       16066     12189             480
financesettlement         9         1               1
internetbanking         156       286              16
jazzvoucher            4619      8472             412
jazzwallet            13505     16933             835
marketingexpense         42         0               1
mcblite                 393       179             137
mygateway                14       652               3
payaxis               22806     61247 

## **Phase 4.1: Statistical Insights & Initial Findings**
### **1. Product & Market Dominance**
* **Best-Selling Categories:** **Mobiles & Tablets** is the clear market leader with **115,701 orders**, followed by **Men's Fashion**.

* **Consumer Preference:** The majority of customers in the Pakistan e-commerce landscape still rely heavily on **Cash on Delivery (COD)**, accounting for over **271,960** transactions.

### **2. Order Health & Fulfillment**
* **Status Distribution:** Out of the entire dataset, **233,658** orders reached **Complete** status, while **201,229** were **Canceled**.

* **Critical Insight:** There is a significant volume of **Canceled** and **Refunded** orders which indicates a need for deeper operational audit.

### **3. Behavioral Correlations (Payment vs. Status)**
* **The COD Paradox:** Contrary to general belief that COD has the most cancellations, it also drives the highest number of completed orders **(148,039 completions)**.

* **Digital Payment Friction:** Systems like easypay and payaxis show a high ratio of cancellations compared to completions **(e.g., easypay: 19,206 Complete vs. 52,840 Canceled)**, suggesting potential technical friction or user mistrust during the digital checkout process.

In [8]:
# Monthly Sales Trend (Orders  Per Month)
print(df['month_name'].value_counts())

# Daily Sales Trend (Orders Per Day of Week)
print(df['day_name'].value_counts())



month_name
November     155438
May           62602
March         61483
August        48511
July          39130
February      38771
June          34521
April         34091
October       30620
December      29197
January       26059
September     24022
Name: count, dtype: int64
day_name
Friday       129364
Thursday      85018
Saturday      79891
Wednesday     77438
Monday        75471
Tuesday       74600
Sunday        62663
Name: count, dtype: int64


In [14]:
# Checking Correlation of 'Month and Category'
month_cat_corr = pd.crosstab(df['month_name'], df['category_name_1'])
month_cat_corr

category_name_1,\n,appliances,beauty & grooming,books,computing,entertainment,health & sports,home & living,kids & baby,men's fashion,mobiles & tablets,others,school & education,soghaat,superstore,unknown,women's fashion
month_name,,,,,,,,,,,,,,,,,
April,179,1876,2138,105,655,656,1455,1676,1145,5205,5756,289,222,1495,6219,100,4920
August,378,2867,4464,249,1308,2143,1343,1859,1449,7774,8557,225,513,3269,6534,1,5578
December,299,2610,2964,114,766,623,992,1408,1226,5771,5254,192,235,2656,935,0,3152
February,202,3239,1786,92,871,2805,1191,1932,1120,5736,8862,4126,214,1357,1777,13,3448
January,343,1870,2170,133,663,752,992,1087,919,3618,5737,218,239,2367,1964,13,2974
July,887,1838,4383,167,1099,1045,1498,1747,1600,4195,9321,305,469,3040,4295,1,3240
June,268,2385,2674,155,707,937,1352,1364,1432,5710,6577,386,178,2001,4950,1,3444
March,227,4346,2513,146,1319,2748,1526,1846,1180,5992,10205,19385,382,1522,1538,5,6603
May,322,7612,3067,223,1710,4155,2562,2370,1812,8981,13227,461,166,3151,5001,30,7752


## **Phase 4.2: Temporal & Seasonality Insights**
### **1. Monthly Peak (The Sales Spike)**
* **November Dominance:** **November** is the highest performing month with **155,438** orders, likely driven by mega sales events like 11.11.

* **Consistency:** Months like **May** and **March** also show strong performance with over **60,000** orders each.

### **2. Weekly Behavioral Patterns**
* **Friday Rush:** **Friday** is the most active shopping day of the week with **129,364** orders.

* **Weekend Dip:** Interestingly, **Sunday** shows the lowest activity **(62,663 orders)**, suggesting users might prefer browsing on weekdays or **Fridays** for deals.

### **3. Category vs. Month Correlation**
* **Mobiles & Tablets in November:** A massive **31,241** orders for **Mobiles & Tablets** were recorded in **November** alone.

* **Fashion Trends:** **Men’s Fashion** peaked in **November** with **32,075** orders, showing that seasonal discounts heavily drive fashion sales.

In [19]:
# Checking Relationship between 'Price' and 'Payment Method'
# Checking the 'Payment Method' of top 10 most expensive items.
top_expensive_orders = df.nlargest(10, 'price')[['price', 'category_name_1', 'payment_method', 'status']]
top_expensive_orders

,price,category_name_1,payment_method,status
509019,1012625.9,beauty & grooming,jazzvoucher,canceled
309752,515975.0,entertainment,payaxis,canceled
207273,479000.0,appliances,easypay,canceled
207344,479000.0,appliances,easypay,canceled
251146,479000.0,appliances,easypay,canceled
251147,479000.0,appliances,payaxis,order_refunded
124009,330499.0,entertainment,payaxis,canceled
182893,330499.0,entertainment,easypay,canceled
423712,320000.0,entertainment,easypay,canceled
341418,307970.0,computing,easypay_ma,canceled


## **Phase 4.4: The Premium Transaction Paradox (Critical Findings)**
### **1. High-Value Order Failure Rate**
* **Zero Success Rate:** Analysis of the **top 10** most expensive items **(ranging from 307,970 to 1,012,625 PKR)** shows a **0% completion rate**.

* **Cancellation Trend:** **9** out of **10** premium orders were **Canceled**, mostly involving payment methods like **Payaxis**, **Easypay**, and **Jazzvoucher**.

* **Refund Issue:** The only order that moved past cancellation worth **479,000 PKR** ended up being **Refunded**, indicating potential issues with high-ticket delivery or customer dissatisfaction.

### **2. Business Implications**
* **Security & Trust:** The high cancellation rate on expensive items (Appliances & Entertainment) suggests that either the payment gateways have strict limits for large amounts or customers are hesitant to commit to such large digital transactions.

* **Operational Leakage:** Processing these high-value "Ghost Orders" puts a strain on inventory management and operational resources without generating actual revenue.

In [22]:
df.to_csv('Pakistan_Ecommerce_Dataset_for_Visualization.csv')
print('File Saved Successfully.')

File Saved Successfully.
